### DISCUSSION TITLE
Policy inheritance for relation_scope

### DISCUSSION THREAD

**Discussion by danielnc** [palkan/action_policy#216] [Q&A] [answered]
I have a base account policy:

```ruby
class Accounts::BasePolicy < ApplicationPolicy
  authorize :user, :account
  relation_scope do |relation|
    relation.where(account_id: account.account_id).order(:id)
  end
```

All my policies inherit from this parent policy so we can properly handle multi tenancy and avoid issues with records leaking across different accounts, but the problem comes when I have a specific policy that needs to add a new relation_scope

```ruby
class Accounts::Products::SectionPolicy < Accounts::BasePolicy
  authorize :product

  relation_scope do |relation|
    relation.where(product_id: product.id).order(:id)
  end
```

authorize is inherited as expected(section policy will require user, account and product)

But its overwriting relation_scope, i would like to have a "default_scope" for inherited policies, how can I properly achieve that where there are no room for human errors when someone overwrites the relation_scope and forgets to add the account_id filter?

**Comment by palkan** [comment, marked as answer]
> how can I properly achieve that where there are no room for human errors when someone overwrites the relation_scope and forgets to add the account_id filter?

You mean something like scopes chaining? That is, the `relation` passed to the child class block should be transformed by the default block?

Then you can use `super`:

```ruby
class Accounts::Products::SectionPolicy < Accounts::BasePolicy
  authorize :product

  relation_scope do |relation|
    super(relation).where(product_id: product.id).order(:id)
  end
end
```

Added an example to the docs: https://github.com/palkan/action_policy/commit/d8c76db9a220138dfbf4b809c17b063e157081a9

**Comment by danielnc** [comment]
Exactly what I was looking for <3
